### PR TITLE
Implement DefaultObjectAcl support in testbench.

### PR DESCRIPTION
**Pull request by coryan**
This will be used in the DefautlObjectAccessControls CRUD APIs,
such as #836.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/984)

### CODE REVIEW

File: google/cloud/storage/tests/testbench.py [hunk 1]
```diff
@@ -448,7 +447,76 @@ def get_acl(self, entity):
...
+        :param entity:str the name of the entity to insert.
+        :param role:str the new role
+        :return:dict the dictionary representing the new ObjectAccessControl.
+        """
+        entity = canonical_entity_name(entity)
+        email = ''
+        if entity.startswith('user-'):
+            email = email.replace('user-', '', 1)
+        # Replace or insert the entry.
+        indexed = {
```
Comment (houglum) [inline google/cloud/storage/tests/testbench.py]: I'm assuming this is supposed to be a dict mapping `str` -> `list(str)`?  For the value in this key-value pair, you forgot the surrounding square brackets... so this actually exhibits some surprising behavior, in that it only sets the key equal to the last item in the iterable:

```
>>> mylist = [1,2,3,4,5]

# Sets mykey to the the last item in the list, 5
>>> mydict = {'mykey': entry for entry in mylist}
>>> mydict
{'mykey': 5}

# I'm assuming this is what you intended to happen here?
>>> mydict = {'mykey': [entry for entry in mylist]}
>>> mydict
{'mykey': [1, 2, 3, 4, 5]}
```

Also, if possible, please try to keep each list comprehension to one line (helps prevent bracket/parenthesis mismatches as shown above).

File: google/cloud/storage/tests/testbench.py [hunk 2]
```diff
@@ -448,7 +447,76 @@ def get_acl(self, entity):
...
+    def delete_default_object_acl(self, entity):
+        """
+        Delete a single default ObjectAccessControl entry from this bucket.
+
+        :param entity:str the name of the entity.
+        :return:None
+        """
+        entity = canonical_entity_name(entity)
+        indexed = {
+            acl.get('entity').lower(): acl
```
Comment (houglum) [inline google/cloud/storage/tests/testbench.py]: Same comment as above.  Actually, this happens in several places in testbench.py.  I did a lazy search for a large number of spaced followed by "for" and found another 2 places:

````
"          for"
````

File: google/cloud/storage/tests/testbench.py (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/tests/testbench.py]: > I'm assuming this is supposed to be a dict mapping `str` -> `list(str)`?

No, this is supposed to be a dict mapping `str` -> `dict`.  I have a list of `dict`, and I want to turn that into a `dict` of `str` to `dict` where the values are the original elements, and the keys are the value of the `entity` element in each `dict`:

```python
>>> mylist = [{'entity': 'a', 'role': 1}, {'entity': 'b', 'role': 2}]
>>> {m['entity']:m for m in mylist}
{'a': {'role': 1, 'entity': 'a'}, 'b': {'role': 2, 'entity': 'b'}}
```

Let me see if I can indent this better (and keep yapf from reformatting it afterwards).

**Comment by coryan** [comment]
PTAL.

**Pull request merged**